packages/modal/src/Modal/Modal.tsx

```diff
@@ -0,0 +1,102 @@
+import * as Dialog from "@radix-ui/react-dialog";
+import { Box, Stack } from "@telegraph/layout";
+import clsx from "clsx";
+import { AnimatePresence, motion } from "framer-motion";
+import React from "react";
+
+type RootProps = Omit<
+  React.ComponentPropsWithoutRef<typeof Dialog.Root>,
+  "modal"
+> &
+  React.ComponentPropsWithoutRef<typeof Stack> & {
+    portal?: {
+      enabled?: boolean;
+      container?: HTMLElement;
+    };
+  };
+
+type RootRef = React.ElementRef<typeof Dialog.Content>;
+
+const Root = React.forwardRef<RootRef, RootProps>(
+  (
+    { portal = { enabled: true }, open, onOpenChange, className, ...props },
+    forwardedRef,
+  ) => {
+    const Portal = portal.enabled ? Dialog.Portal : React.Fragment;
+    return (
+      <Dialog.Root open={open} onOpenChange={onOpenChange}>
+        <AnimatePresence>
```

Comment:
Solid detail ✨

Probably worth watching how we use animation going forward and if/when we want to make this customizable and/or reusable